### PR TITLE
Add Quick Reference Marker

### PR DESCRIPTION
**Pull request by JoramD0**
**When merged this pull request will:**
- Add Quick Reference Marker keybind/functionality

Quick Reference Markers (QRMs) allow pilots to quickly make a marker on their MicroDAGR of a point of interest.

### CODE REVIEW

File: addons/ratelmarker/stringtable.xml [hunk 1]
```diff
@@ -25,5 +25,23 @@
             <German>RATEL Markierung erstellt (Karte öffnen um zu zentrieren)</German>
             <French>Balise RATEL crée (Ouvrir la carte pour focaliser)</French>
         </Key>
+        <Key ID="STR_TAC_RatelMarker_QRM_settingsCategory">
+            <English>TAC Quick Reference Marker</English>
```
Comment (jonpas) [inline addons/ratelmarker/stringtable.xml:29]: Make a new "Markers" category instead and keep them together. Maybe split with sub-categories.

File: addons/ratelmarker/stringtable.xml [hunk 2]
```diff
@@ -25,5 +25,23 @@
...
+        </Key>
+        <Key ID="STR_TAC_RatelMarker_QRM_Disabled">
+            <English>Disabled</English>
```
Comment (jonpas) [inline addons/ratelmarker/stringtable.xml:32]: Can use `STR_ACE_Common_Disabled`.

File: addons/ratelmarker/functions/fnc_prepareQRM.sqf [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+if (GVAR(QuickReferenceMarkerMode) == 2) then {
+    private _traceBegin = AGLToASL positionCameraToWorld [0, 0, 0];
+    private _traceEnd = AGLToASL positionCameraToWorld [0, 0, 5000];
+
+    private _markerPos = terrainIntersectAtASL [_traceBegin, _traceEnd];
+
+    if !(_markerPos isEqualTo [0,0,0]) then {
+        [_markerPos] call FUNC(createQRM);
+    } else {
+        "Could not create Quick Reference Marker" call CBA_fnc_notify;
```
Comment (jonpas) [inline addons/ratelmarker/functions/fnc_prepareQRM.sqf:37]: Stringtable.

File: addons/ratelmarker/functions/fnc_prepareQRM.sqf [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    [_markerPos] call FUNC(createQRM);
+};
+
+if (GVAR(QuickReferenceMarkerMode) == 2) then {
+    private _traceBegin = AGLToASL positionCameraToWorld [0, 0, 0];
+    private _traceEnd = AGLToASL positionCameraToWorld [0, 0, 5000];
+
+    private _markerPos = terrainIntersectAtASL [_traceBegin, _traceEnd];
+
+    if !(_markerPos isEqualTo [0,0,0]) then {
```
Comment (jonpas) [inline addons/ratelmarker/functions/fnc_prepareQRM.sqf:34]:
```suggestion
    if !(_markerPos isEqualTo [0, 0, 0]) then {
```

File: addons/ratelmarker/functions/fnc_createQRM.sqf [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * Public: No
+ */
+
+params ["_markerPos"];
+
+[
+    format ["QRM [%1] (%2)", (mapGridPosition _markerPos), [daytime, "HH:MM"] call BIS_fnc_timeToString],
+    _markerPos
+] call ace_microdagr_fnc_deviceAddWaypoint;
```
Comment (jonpas) [inline addons/ratelmarker/functions/fnc_createQRM.sqf:23]:
```suggestion
] call ACEFUNC(microdagr,deviceAddWaypoint);
```

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/ratelmarker/XEH_preInit.sqf [hunk 6]
```diff
@@ -7,3 +7,11 @@ PREP_RECOMPILE_START;
 PREP_RECOMPILE_END;
 
 ADDON = true;
+
+[
+    QGVAR(QuickReferenceMarkerMode),
+    "LIST",
+    LSTRING(QRM_Mode),
+    LSTRING(QRM_settingsCategory),
+    [[0, 1, 2], [LSTRING(QRM_Disabled), LSTRING(QRM_VehiclePos), LSTRING(QRM_ViewPos)], 2]
```
[the remark this answers is not in the document]
Comment (JoramD0) [reply addons/ratelmarker/XEH_preInit.sqf:16]: I'd like to have it as an in-game option if we ever decide it is overpowered or something ridiculous.

**Pull request merged**